parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -107,6 +109,20 @@ def __init__(self,
 
         logger.info("Manager started")
 
+        try:
+            ix_address = probe_addresses(addresses.split(','), task_port)
+            if not ix_address:
+                logger.error("Unable to connect back on any addresses:{}".format(addresses))
+                exit(5)
+            else:
+                logger.info("Connection to Interchange successful on {}".format(ix_address))
+                task_q_url = "tcp://{}:{}".format(ix_address, task_port)
+                result_q_url = "tcp://{}:{}".format(ix_address, result_port)
+                logger.info("Task url : {}".format(task_q_url))
+                logger.info("Result url : {}".format(result_q_url))
+        except Exception:
+            logger.exception("Caught exception while trying to determine viable address to interchange")
```

Comment:
should the process worker terminate now if this exception path happens? (because the iniitalisations haven't necessarily all been run correctly)

Reply:
Right, I think we should terminate here if the probe fails.

Reply:
ok. other than this, this PR is ready to be approved